lib/dropsonde/cache.rb

```diff
@@ -1,34 +1,37 @@
+# frozen_string_literal: true
+
 require 'date'
 require 'json'
 require 'fileutils'
 require 'puppet_forge'
 
+# cache class
 class Dropsonde::Cache
-  @@autoupdate = false
+  @autoupdate = false
 
-  def self.init(path, ttl, autoupdate)
+  def initialize(path, ttl, autoupdate)
     FileUtils.mkdir_p(path)
-    @@path = "#{File.expand_path(path)}/forge.json"
-    @@ttl  = ttl
-    @@autoupdate = autoupdate
-
-    if File.file? @@path
-      @@cache = JSON.parse(File.read(@@path))
-    else
-      @@cache = {
-                'timestamp' => '2000-1-1',  # long before any puppet modules were released!
-                'modules'   => [],
-              }
-    end
-
-    PuppetForge.user_agent = "Dropsonde Telemetry Client/0.0.1"
+    @path = "#{File.expand_path(path)}/forge.json"
+    @ttl  = ttl
+    @autoupdate = autoupdate
+
+    @cache = if File.file? @path
+               JSON.parse(File.read(@path))
+             else
+               {
+                 'timestamp' => '2000-1-1', # long before any puppet modules were released!
+                 'modules' => [],
+               }
+             end
+
+    PuppetForge.user_agent = 'Dropsonde Telemetry Client/0.0.1'
   end
 
-  def self.modules
-    @@cache['modules']
+  def modules
+    @cache['modules']
```

Comment:
I finally figured out that this corresponds to the change in https://github.com/puppetlabs/dropsonde/pull/5/files#diff-970826421b598051d54606187ccb9e5b33f7d16b3ab0fe90f0780b211e12c5b7R41 ; it would be easier to understand if such fundamental changes would be in a separate commit next time, so they're not obscured by the rest of the rubocop changes.

Reply:
I pushed a new commit to use `securerandom` instead of previous function.

Reply:
here is the [commit](https://github.com/puppetlabs/dropsonde/pull/5/commits/c8bd5fe0b13e1257b5f63adb7627f4bf030b0a67)